pkg/i2gw/providers/istio/resource_reader.go

```diff
@@ -44,6 +45,12 @@ func newResourceReader(conf *i2gw.ProviderConf) reader {
 func (r *reader) readResourcesFromCluster(ctx context.Context) (*storage, error) {
 	res := newResourcesStorage()
 
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	if err := r.conf.Client.Get(ctx, types.NamespacedName{Name: CRDName}, crd); err != nil {
+		log.Printf("CRD '%s' is not installed in the cluster, will not continue reading resources", CRDName)
+		return res, nil
+	}
+
```

Comment:
Sorry this was not added to the previous review. 
The intention was to change the call to read ANY CRD. In this case you only cover Istio Gateway but it is also relevant for VirtualServices and any other provider's CRD that will be added in the future.

So either you change istio calls to "fail soft" or you create a general helper that doing it